sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/FluxAutoRenew.java

```diff
@@ -0,0 +1,164 @@
+            LockContainer<LockRenewalOperation> messageLockContainer,
+            Function<String, Mono<OffsetDateTime>> onRenewLock) {
+            this.onRenewLock = Objects.requireNonNull(onRenewLock, "'onRenewLock' cannot be null.");
+            this.actual = actual;
+            this.messageLockContainer = Objects.requireNonNull(messageLockContainer,
+                "'messageLockContainer' cannot be null.");
+            Objects.requireNonNull(maxAutoLockRenewDuration, "'maxAutoLockRenewDuration' cannot be null.");
+            this.maxAutoLockRenewal = maxAutoLockRenewDuration;
+        }
+        /**
+         * On an initial subscription, will take the first work item, and request that amount of work for it.
+         * @param subscription Subscription for upstream.
+         */
+        @Override
+        protected void hookOnSubscribe(Subscription subscription) {
+            Objects.requireNonNull(subscription, "'subscription' cannot be null.");
+            actual.onSubscribe(subscription);
+        }
+
+        /**
+         * When upstream has completed emitting messages.
+         */
+        @Override
+        public void hookOnComplete() {
+            logger.info("Upstream has completed.");
+            actual.onComplete();
+        }
+
+        @Override
+        protected void hookOnError(Throwable throwable) {
+            logger.info("Errors occurred upstream.", throwable);
+            actual.onError(throwable);
+            dispose();
+        }
+
+        @Override
+        protected void hookOnNext(ServiceBusReceivedMessage message) {
+            final String lockToken = message.getLockToken();
+            final OffsetDateTime lockedUntil = message.getLockedUntil();
+            final LockRenewalOperation renewOperation;
+
+            if (Objects.isNull(lockToken)) {
+                logger.info("Unexpected, LockToken is not present in message [{}].", message.getSequenceNumber());
+                return;
+            } else if (Objects.isNull(lockedUntil)) {
+                logger.info("Unexpected, lockedUntil is not present in message [{}].", message.getSequenceNumber());
+                return;
+            }
+
+            final Function<String, Mono<OffsetDateTime>> onRenewLockUpdateMessage = onRenewLock.andThen(updated ->
+                updated.map(newLockedUntil -> {
+                    message.setLockedUntil(newLockedUntil);
+                    return newLockedUntil;
+                }));
+
+            renewOperation = new LockRenewalOperation(lockToken, maxAutoLockRenewal, false,
+                onRenewLockUpdateMessage, lockedUntil);
+
+            try {
+                messageLockContainer.addOrUpdate(lockToken, OffsetDateTime.now().plus(maxAutoLockRenewal),
+                    renewOperation);
+            } catch (Exception e) {
+                logger.error("Exception occurred while updating lockContainer for token [{}].", lockToken, e);
```

Comment:
Don't use `error` level here as we are not reporting these errors to the user. It will show up in their log and can alert them.